### ISSUE TITLE
Intermittent scroll interruption during scrolling

### ISSUE THREAD

**Issue by ramb6** [Caldis/Mos#410]
- 我们不想将反馈问题的门槛抬得太高, 所以你不需要完全将下方的内容填完, 但是, 提供尽可能完整的内容可以让我们更快速地定位问题
- We don't want to make the feedback problem too complicated, so you don't need to completely fill out the content below, but providing content that is as complete as possible allows us to locate problems more quickly.

---

  感谢您的反馈, 请在提交前将上方文字删去 | Thanks for your contribution, please delete the text above before submitting

---

- 问题描述 | Issue description:
Using a 3rd party (logitech g502) mouse and while scrolling (so far only noticeable in native mac os apps, such as messages)

- 如何重现 | How to reproduce it:
With MOS enabled (at any settings), while quick mouse scrolling, the scrolling will stop even if the user is physically scrolling the scroll wheel. It will continue to scroll only after a user has stopped scrolling and restarts the process, when it will happen again.

So far this can be reproduced consistently in the messages app. So far the solution has been to disable MOS through exceptions in the MOS preferences. 

- 期望结果 | What is expected:

Scrolling stopping when unintended in native mac os apps such as messages.

Even if adjusting the scroll speed under Mouse preferences in System preferences, the behavior remains consistent and equally unusable. 
- 实际结果 | What is actually happening:

scrolling stops even when still scrolling physically on the mouse. leads to unintended behavior and reduces usability and functionality. 
- 软件版本 | Application version:
14.0 (6000)

- 系统版本 | System version:
11.4

- 您系统中的其他鼠标管理软件 | Other mouse management application in your system:

None. 

Logitech is known for G hub software for their mice, however it has low functionality and consistency on mac os, and therefore is not used or installed in this scenario.

**Comment by hrvstr** [comment]
I think this is a duplicate of my issue #379. (correct me if I'm wrong)

**Comment by ramb6** [comment]
Yes, it seems to be. It’s multiple app issue, both system apps 3rd party.

**Comment by pkrayzy** [comment]
I think this is an 'issue' with ScrollPhase. I am not sure if it breaks other things (I haven't noticed anything working unexpectedly since doing so), but setting all the Double values below to 0.0 and rebuilding has seemed to correct this issue for me.

`let PhaseValueMapping: [Phase: [PhaseItem: Double]] = [
    // 空
    Phase.Idle: [PhaseItem.Scroll: 0.0, PhaseItem.Momentum: 0.0],
    // 手指触碰
    Phase.Contact: [PhaseItem.Scroll: 128.0, PhaseItem.Momentum: 0.0],
    // 跟随滚动
    Phase.Tracing: [PhaseItem.Scroll: 2.0, PhaseItem.Momentum: 0.0],
    // 缓动
    Phase.Momentum: [PhaseItem.Scroll: 0.0, PhaseItem.Momentum: 2.0],
    // 缓动停止 (自动, 如缓动到达临界值)
    Phase.PauseAuto: [PhaseItem.Scroll: 0.0, PhaseItem.Momentum: 3.0],
    // 缓动停止 (手动)
    Phase.PauseManual: [PhaseItem.Scroll: 4.0, PhaseItem.Momentum: 0.0],
]`

**Comment by Desousak** [comment]
After some testing I was able to 'fix' the scrolling by setting `PhaseItem.Scroll` within `Phase.PauseManual` to `0.0`. So far I haven't noticed any unexpected behaviour.

Here's a build of Mos from the `dev` branch with the mentioned change made: [Mos - Fixed Catalyst Scrolling.zip](https://github.com/Caldis/Mos/files/8233661/Mos.-.Fixed.Catalyst.Scrolling.zip) (or you can build it yourself)

**Comment by hrvstr** [comment]
> After some testing I was able to 'fix' the scrolling by setting `PhaseItem.Scroll` within `Phase.PauseManual` to `0.0`. So far I haven't noticed any unexpected behaviour.
> 
> Here's a build of Mos from the `dev` branch with the mentioned change made: [Mos - Fixed Catalyst Scrolling.zip](https://github.com/Caldis/Mos/files/8233661/Mos.-.Fixed.Catalyst.Scrolling.zip) (or you can build it yourself)

Can confirm. These changes seem to have fixed the issue for me, too. Nice work!

**Comment by kcstagepro** [comment]
https://github.com/Caldis/Mos/issues/333 

i bought a 8gb Mac mini about a month ago - was getting a similar lag issue to #333, even though that seems resolved 

so i ordered a pair of 16gb mac mini's one for home one for office - same MX vertical mouse - same mac mini same everything but different display

office machine works fine - home has been giving me lag at the start of some scrolls, much like #333   

i think this [Mos - Fixed Catalyst Scrolling.zip](https://github.com/Caldis/Mos/files/8233661/Mos.-.Fixed.Catalyst.Scrolling.zip)
 might have fixed it

**Comment by hrvstr** [comment]
> > After some testing I was able to 'fix' the scrolling by setting `PhaseItem.Scroll` within `Phase.PauseManual` to `0.0`. So far I haven't noticed any unexpected behaviour.
> > Here's a build of Mos from the `dev` branch with the mentioned change made: [Mos - Fixed Catalyst Scrolling.zip](https://github.com/Caldis/Mos/files/8233661/Mos.-.Fixed.Catalyst.Scrolling.zip) (or you can build it yourself)
> 
> Seems like this issues got fixed in some other way. I compiled the current source code without your changes and with that build the scrolling issues in Messages are gone, even though PhaseItem.Scroll for Phase.PauseManual is at 4.0.

Can you share the compiled App? I don't want to install Xcode just to get the latest version.

**Comment by Diskutant** [comment]
Not sure if it will work due to signing stuff and all.
But here it is: [Mos.zip](https://github.com/Caldis/Mos/files/11745633/Mos.zip)

**Comment by hrvstr** [comment]
Thanks, working fine once I allowed it to bypass the Gatekeeper. Scrolling in the Twitter app is really smooth now!
